text/0693-property-injection.md

```diff
@@ -0,0 +1,468 @@
+
+## Working Backwards
+
+### Development Team Experience
+
+Development teams can start using Blueprint by attaching propertyInjectors to App.
+In the example below, this dev team wants use the Property Injectors provided by its org for S3 Buckets and Lambda Fucntions. 
+
+```ts
+import { BucketPropsInjector, FunctionPropsInjector } from '@my-org/standard-props-injectors';
+...
+const app = new App({
+  ...
+  propertyInjectors: [
+    new BucketPropsInjector(),
+    new FunctionPropsInjector(),
+  ],
+});
+```
+
+Dev teams can also attach propertyInjectors to Stacks as well.  This will override the Bucket Injector provided in App.
+
+```ts
+import { BetaBucketPropsInjector } from '@my-team/custom-props-injectors';
+...
+const betaStack = new Stack(stage, 'MyApp-beta-stack', {
+  env: { account: '123456789012', region: 'us-east-1' },
+  propertyInjectors: [
+    new BetaBucketPropsInjector(),
+  ],
+});
+```
+
+In this case, Constructs created in betaStack will use `BetaBucketPropsInjector` instead of `BucketPropsInjector`.  This is useful when Beta resources have different requirements. 
+
+See [Scope Tree Traversal](#scope-tree-traversal) for more information.
+
+### Organization Level Standardization
+
+As an org with many dev teams, we can standardize how AWS Resources are created by providing dev teams with Injectors.
+
+AWS CDK will expose this `IPropertyInjector` interface for orgs to implement:
+
+```ts
+/**
+ * This defines the context ofInjection.
+ */
+export interface InjectionContext {
+  /**
+   * scope from the Construct constructor
+   */
+  readonly scope: Construct;
+  /**
+   * id from the Construct constructor
+   */
+  readonly id: string;
+}
+
+/**
+ * This attaches the inject function to the Construct that it operates on.
+ */
+export interface IPropertyInjector {
+  /**
+   * The FQN of the Construct class.
+   */
+  readonly constructFqn: string;
+
+  /**
+   * The injector to be applied to the constructor of the Construct.
+   */
+  inject(originalProps: any, context: InjectionContext): any;
+}
+```
+
+Here is a simple PropertyInjector to make sure the S3 Bucket is not publically accessible and SSL is used.
+
+```ts
+class MyBucketPropsInjector implements IPropertyInjector {
+  public readonly constructFqn: string;
+
+  constructor() {
+    this.constructFqn = Bucket.UNIQUE_FQN;
+  }
+
+  inject(originalProps: BucketProps, _context: InjectionContext): BucketProps {
+    return {
+      blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+      enforceSSL: true,
+      ...originalProps,
+    };
+  }
+}
+```
+
+Notes:
+* We will add Bucket.UNIQUE_FQN to unique identify this Construct.  This is a new Property that we will add to Bucket and other supported Constructs.
+* This TypeScript code will set blockPublicAccess to BlockPublicAccess.BLOCK_ALL and enforceSSL to true if these properties are not specified in originalProps.  This implementation allows the dev teams to override the org's recommended defaults.
```

Comment:
Also include a sentence here making explicit that it's the exmaple implementation of the injector that leads to dev teams being able to overwrite the defaults. The injector can be written differently to explicitly always overwrite dev team decisions.